content/articles/how-to-add-support-for-spatial-data-in-your-database-using-geoalchemy-and-spatialite/index.md

```diff
@@ -0,0 +1,389 @@
+
+The site also allows you to download this as an excel spreadsheet, so head over to the website to get your own list of random lon/lat values. I had to do some editing on the file to remove some unneeded columns and fields, in order for it to look like the image shown above, I also saved it as a csv file instead into the root level of our project directory for ease of access.
+
+ So I'm going to use pandas to parse this file. We can do this from a new file called `data.py`, create this file in the root level of your project folder on the same level as `wsgi.py`. If you don't have `Pandas` installed, you can install it using the pip command as follows:
+
+```
+pip install pandas
+```
+
+In the `data.py` file add the following lines of code to it to feed data from the spreadsheet (or csv file):
+
+```python
+from wsgi import app
+from core import db
+from models import *
+import pandas as pd
+import random
+
+with app.app_context():
+    data_ = pd.read_csv('sample_locations.csv')
+    df = pd.DataFrame(data_)
+    names = ["Joe", "Mike", "Paul"]
+    for index, row in df.iterrows():
+        new_user = User(
+            name=names[random.randint(0,2)],
+            lon=row.X,
+            lat=row.Y,
+            is_handyman=bool(random.randint(0,1)),
+            geometry='SRID=4269;POINT(%.8f %.8f)'%(row.X, row.Y)
+        )
+        db.session.add(new_user)
+    db.session.commit()
+```
+Now we have data we can query and work with. Let's say we need to find if there are any handymen located within a 10,000meter radius to a certain user located at `POINT(145.67, -30.513)`, we can make a spatial query to the database, from `data.py` as follows:
+
+```python
+from sqlalchemy.sql import func
+# ..
+
+with app.app_context():
+    lng, lat = 148.523, -35.40
+    geo_wkb = func.ST_PointFromWKB(User.query.all()[0].geometry)
+
+    new_point = db.session.query(User).filter(
+        func.PtDistWithin(
+            User.geometry,
+            geo_wkb,
+            10000
+        )
+    ).one()
+    print(new_point)
+```
+Typically, you might want to add these to your api-endpoints or have them as functions that your endpoints can call and serve you the required information. Whichever case you want, you can surely achieve using these steps.
+
+## Conclusion
+
+In this tutorial, we have seen how to set up and configure a basic web application, that utilizes Spatialite and sqlite to perform read and write operations to a spatial database. I covered how to install spatiallite, and also showed how to update your existing instance of sqlite to support spatial data. Now you are fully equipped to set-up your development environment for your projects using spatialite and geoalchemy2. 
+
+**Note**: While sqlite3 is great for your development environment, it is advised to not use it for heavy duty applications, you might instead want to consider using a more sophisticated tool like PostgresQL, and PostGIS as an extension for it's support of spatial data.
```

Comment:
Add a section called `Further reading` that provides other reference links

Reply:
done